packages/core/components/EditMetadata/NewAnnotationPathway.tsx

```diff
@@ -18,21 +31,87 @@ enum EditStep {
 
 interface NewAnnotationProps {
     onDismiss: () => void;
-    onUnsavedChanges: () => void;
+    hasUnsavedChanges: (arg: boolean) => void;
```

Comment:
`hasUnsavedChanges` this is the naming convention of a boolean so it feels weird here to me - it seems like this is just the same thing as `onDismiss` but has an argument so would it make sense to add a parameter of "hasUnsavedChanges: boolean" to `onDismiss`?